### ISSUE TITLE
Expand documentation to include guidance on injecting `HttpClient` directly in controllers, view components and tag helpers

### ISSUE THREAD

**Issue by chrisdpratt** [aspnet/AspNetCore.Docs#8418]
I was recently involved in an interesting discussion on StackOverflow, related to a [question](https://stackoverflow.com/questions/52186856/unable-to-resolve-service-for-type-system-net-http-httpclient?noredirect=1#comment91330222_52186856) where the OP was attempting to inject `HttpClient` in a view component's constructor. Even though the view component was properly registered with a typed client in Startup.cs, an exception was still being thrown during injection.

Turns out that even though view components can be dependency injected in general (including even `IHttpClientFactory`), you cannot inject `HttpClient` without adding:

    services.AddMvc().AddViewComponentsAsServices();

The same goes for controllers and tag helpers, which have their own `.AddControllersAsServices()` and `.AddTagHelpersAsServices()` extensions, respectively. The documentation here is completely silent on this.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 881cef77-11ef-ac11-06d0-0996a1ccfa1b
* Version Independent ID: 3d356334-e672-7e8b-1fef-7944ebee99eb
* Content: [Initiate HTTP requests](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/http-requests?view=aspnetcore-2.1)
* Content Source: [aspnetcore/fundamentals/http-requests.md](https://github.com/aspnet/Docs/blob/master/aspnetcore/fundamentals/http-requests.md)
* Product: **aspnet-core**
* GitHub Login: @stevejgordon
* Microsoft Alias: **scaddie**

**Comment by Rick-Anderson** [comment]
@serpent5 Can you review/help me if I get stuck on this one?

**Comment by serpent5** [comment]
@Rick-Anderson I'd be very happy to help, but in this case I no longer think this is something that should be documented. I reproduced the issue in 2.2, just in case it had been "fixed" there, but I then realised that using a View Component, Tag Helper or Controller as a *typed client* actually just doesn't make a lot of sense. I did also dig into *why* DI doesn't work for it out of the box, which I might write up into the existing answer, but I think it's all intentional and might be better left alone.

Let me know if you'd like me to go into more detail on it all here, but I would fully expect that the PU team would agree that this just isn't a sensible scenario and isn't something they'd want to encourage.
